clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -217,7 +245,7 @@ private boolean isTransitionValid(State source, State target) {
                 case UNINITIALIZED:
                     return source == READY;
                 case INITIALIZING:
-                    return source == UNINITIALIZED;
+                    return source == UNINITIALIZED || source == ABORTING_TRANSACTION;
```

Comment:
Below we use the InitProducerId API as another way to abort a transaction. Rather than transitioning to `INITIALIZING` in this case, I wonder if it makes sense to stay in `ABORTING_TRANSACTION`?